code/game/objects/items/stacks/stack.dm

```diff
@@ -172,13 +172,31 @@
 
 /obj/item/stack/proc/use(var/amount)
 	ASSERT(isnum(src.amount))
+
 	if(src.amount>=amount)
 		src.amount-=amount
 	else
 		return 0
 	. = 1
-	if (src.amount<=0)
+	if (src.amount<=0) //If the stack is empty after removing the required amount of items!
 		if(usr)
+			if(istype(usr,/mob/living/silicon/robot))
+				var/mob/living/silicon/robot/R=usr
+				if(R.module)
```

Comment:
But I can't see why it would be.
This exact code is executed in before_take_item